pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1892,7 +1873,35 @@ func nodeUsingCSIPlugin(og OperationGenerator, spec *volume.Spec, nodeName types
 	}
 
 	return isMigratedOnNode, nil
+}
+
+func nodeUsingCSIPlugin(og OperationGenerator, spec *volume.Spec, nodeName types.NodeName) (bool, error) {
+	vpm := og.GetVolumePluginMgr()
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) ||
+		!utilfeature.DefaultFeatureGate.Enabled(features.CSINodeInfo) {
+		return false, nil
+	}
+	migratable, err := vpm.IsPluginMigratableBySpec(spec)
+	if err != nil {
+		return false, err
+	}
+	if !migratable {
+		return false, nil
+	}
+
+	if len(nodeName) == 0 {
+		return false, goerrors.New("nodeName is empty")
+	}
+
+	kubeClient := vpm.Host.GetKubeClient()
+	if kubeClient == nil {
+		// Don't handle the controller/kubelet version skew check and fallback
+		// to just checking the feature gates. This can happen if
+		// we are in a standalone (headless) Kubelet
+		return true, nil
+	}
+	return checkDriverInstalled(vpm, spec, nodeName)
```

Comment:
this changes behavior significantly in that now if driver is not installed we do not do migration. In fact, when Migration flags are on we always want to do migration whether the driver is installed or not.

Reply:
checkDriverInstalled just refactors existing code - keeping the return values intact.